andy/src/main/java/nl/tudelft/cse1110/andy/result/ResultBuilder.java

```diff
@@ -78,9 +78,9 @@ public void compilationFail(List<Diagnostic<? extends JavaFileObject>> errors) {
         this.compilation = CompilationResult.compilationFail(compilationErrors);
     }
 
-    public void compilationSecurityFail(String message) {
+    public void compilationSecurityFail(String message, Optional<Integer> lineNumber) {
```

Comment:
You have an `Optional` here but you're passing `null` as an argument on line 51 of `SourceCodeSecurityCheckStep.java`. This leads to an exception.

I don't think it's a good idea to use an Optional as a method argument - it's way too easy for someone to accidentally pass `null` instead of `Optional.empty()`.

Reply:
Yeah, that seems like a good idea - you can overload the method:
- `compilationSecurityFail(String message, int lineNumber)`, which does the same as what you have now (except that the optional is replaced with a simple int)
- `compilationSecurityFail(String message)`, which calls `compilationSecurityFail(message, 1)`